Extension/src/ghAPI.ts

```diff
@@ -0,0 +1,152 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+'use strict';
+
+import { PackageVersion } from './packageVersion';
+import * as util from './common';
+import * as tmp from 'tmp';
+import * as telemetry from './telemetry';
+import { PlatformInformation } from './platform';
+
+async function parseJsonAtPath(path: string): Promise<any> {
+    try {
+        const exists: boolean = await util.checkFileExists(path);
+        if (exists) {
+            const fileContent: string = await util.readFileText(path);
+            return JSON.parse(fileContent);
+        }
+    } catch (error) {
+    }
+}
+
+interface Asset {
+    name: string;
+    browserDownloadUrl: string;
+}
+
+interface Build {
+    name: string;
+    assets: Asset[];
+}
+
+function isAsset(input: any): input is Asset {
+    return input && input.name && typeof(input.name) === "string" && input.browser_download_url && input.browser_download_url && typeof(input.browser_download_url) === "string"; 
+}
+
+function isBuild(input: any): input is Build {
+    return input && input.name && typeof(input.name) === "string" && isArrayOfAssets(input.assets) && input.assets.length >= 4;
+}
+
+function isArrayOfAssets(input: any): input is Asset[] {
+    return input instanceof Array && input.every(item => isAsset(item));
+}
+
+function isReleaseJson(input: any): input is Build[] {
+    return input && input instanceof Array && input.length !== 0 && input.every(item => isBuild(item));
```

Comment:
as we discussed, maybe don't validate "every" item, just the first few.  (or maybe validate until you find both a prerelease and stable release?)